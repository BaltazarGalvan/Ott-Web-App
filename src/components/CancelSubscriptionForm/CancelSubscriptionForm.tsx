import React from 'react';
import { useTranslation } from 'react-i18next';

import Button from '../Button/Button';
import FormFeedback from '../FormFeedback/FormFeedback';

import styles from './CancelSubscriptionForm.module.scss';

type Props = {
  onConfirm: () => void;
  onCancel: () => void;
  error: string | null;
};

const CancelSubscriptionForm: React.FC<Props> = ({ onConfirm, onCancel, error }: Props) => {
  const { t } = useTranslation('account');

  return (
    <div>
      {error ? <FormFeedback variant="error">{error}</FormFeedback> : null}
      <h2 className={styles.title}>{t('cancel_subscription.title')}</h2>
      <p className={styles.paragraph}>
        {t('cancel_subscription.explanation')}
      </p>
      <Button className={styles.confirmButton} label="Unsubscribe" color="primary" variant="contained" onClick={onConfirm} fullWidth />
      <Button label="No, thanks" variant="outlined" onClick={onCancel} fullWidth />
    </div>
  );
};

export default CancelSubscriptionForm;
