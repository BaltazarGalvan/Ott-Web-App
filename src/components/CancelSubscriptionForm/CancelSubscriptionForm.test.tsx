import React from 'react';
import { render } from '@testing-library/react';

import CancelSubscriptionForm from './CancelSubscriptionForm';

describe('<CancelSubscriptionForm>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<CancelSubscriptionForm submitting={false} error={null} onCancel={jest.fn()} onConfirm={jest.fn()} />);

    expect(container).toMatchSnapshot();
  });
});
