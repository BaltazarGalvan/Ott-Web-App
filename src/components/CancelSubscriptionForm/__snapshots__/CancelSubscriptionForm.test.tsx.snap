// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CancelSubscriptionForm> renders and matches snapshot 1`] = `
<div>
  <div>
    <h2
      class="title"
    >
      cancel_subscription.title
    </h2>
    <p
      class="paragraph"
    >
      cancel_subscription.explanation
    </p>
    <button
      class="button confirmButton primary fullWidth"
    >
      <span>
        Unsubscribe
      </span>
    </button>
    <button
      class="button default outlined fullWidth"
    >
      <span>
        No, thanks
      </span>
    </button>
  </div>
</div>
`;
